File: 69-sqrtx/sqrtx.py
class Solution(object):
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x==0 or x==1:
            return x
        start=0
        end=x
        while start<=end:
            mid=start+(end-start)//2
            if mid*mid>x:
                 end=mid-1
            elif mid*mid<x:
                start=mid+1
                res=mid
            else:
                return mid
        return res

        

File: 69-sqrtx/sqrtx.java


class Solution {
    public int mySqrt(int x) {
        int st =1;
        int ed =x;
        int mid = st+(ed-st)/2;      

        while(st<=ed){
            if(mid == x/mid){
                return mid;
            }else if(mid>x/mid){
                ed=mid-1;                
            }else{
                st=mid+1;
            }
            mid = st+(ed-st)/2;
        }  
        return ed;      
    }
}